torch/csrc/autograd/engine.cpp

```diff
@@ -33,6 +33,11 @@ namespace torch { namespace autograd {
 static constexpr int NO_DEVICE = -2;
 static thread_local int worker_device = NO_DEVICE;
 
+// this variable captures whether we are inside torch.autograd.grad() or not.
```

Comment:
No, the condition is more complicated than that, right? This variable is true if ALL invocations in the stack of reentrant engine invocations are mutable backwards. Of course the simple situation is good to also give for intuition, but precision is important too.

Reply:
yep, forgot to update the comments after discussion. I'll update this. Thanks